week06/my-app/src/pages/api/post/delete.ts

```diff
@@ -0,0 +1,38 @@
+import { connectDB } from "@/utils/database";
+import { ObjectId } from "mongodb";
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getServerSession } from "next-auth";
+import { authOptions } from "../auth/[...nextauth]";
+
+export default async function handler (
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method !== "DELETE") {
+    return res.status(405).json({ message: "지원하지 않는 메서드입니다" });
+  }
+
+  let session = await getServerSession(req, res, authOptions);
+  if (!session?.user?.email) {
+    return res.status(401).json({ messae: "로그인 필요" });
+  }
+
+  const { _id, author } = req.body;
+
+  const db = (await connectDB).db("forum");
+  const post = await db.collection("post").findOne({ _id: new ObjectId(_id) });
+  if (!post) return res.status(404).json({ message: "게시글 없음" });
```

Comment:
(listItem 코드 설명에 이어서) 그런데 여기서 req.body를 _id와 author라는 '객체'로 받아오고 있습니다. 그래서 실제 req.body에는 문자열 하나만이 담겨있는데, 객체 형태로 값을 받아오고 있으니 req.body._id 는 undefined가 되어 `new ObjectId(_id)` 도 `new ObjectId(undefined)로 호출되게 됩니다. 그러면 DB에 그런 _id가 없기 때문에 post가 null이 되면서 모든 글에 대해 '게시글 없음' 메세지가 출력되고 있는 것으로 보입니다!  (author 역시도 undefined가 됨)

그래서 만약 이 부분 코드만 고친다면 
```suggestion
  const postId = req.body;

  const db = (await connectDB).db("forum");
  const post = await db.collection("post").findOne({ _id: new ObjectId(postId) });
  if (!post) return res.status(404).json({ message: "게시글 없음" });
```

이렇게 바꿔주시면 정상 작동할 것 같습니다. 

author는 post.author로 받아오고 있어서 req.body로 받아올 필요는 없습니당.

Reply:
- **수정방법 2)**
body를 객체로 반환하도록 ListItem을 수정했으므로, 여기서는 기존과 같이 객체로 받아도 문제가 생기지 않게됩니다.

다만 body에서 반환하는 값에 맞춰서 
`const {_id}=req.body` 또는 `const { _id, author } = req.body;`
이렇게 작성하시고 처음 사용하신 것 그대로 ObjectId(_id)로 사용하셔도 됩니다!!